clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -848,6 +853,70 @@ public void testWakeupCommitted() {
         assertNull(consumer.wakeupTrigger().getPendingTask());
     }
 
+    @Test
+    public void testInterceptorAutoCommitOnClose() {
+        Properties props = requiredConsumerPropertiesAndGroupId("test-id");
+        props.setProperty(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, MockConsumerInterceptor.class.getName());
+        props.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+
+        consumer = newConsumer(props);
+        assertEquals(1, MockConsumerInterceptor.INIT_COUNT.get());
+        completeCommitApplicationEventSuccessfully();
+
+        consumer.close(Duration.ZERO);
+
+        assertEquals(1, MockConsumerInterceptor.ON_COMMIT_COUNT.get());
+        assertEquals(1, MockConsumerInterceptor.CLOSE_COUNT.get());
+    }
+
+    @Test
+    public void testInterceptorCommitSync() {
+        Properties props = requiredConsumerPropertiesAndGroupId("test-id");
+        props.setProperty(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, MockConsumerInterceptor.class.getName());
+        props.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+        consumer = newConsumer(props);
+        assertEquals(1, MockConsumerInterceptor.INIT_COUNT.get());
+        completeCommitApplicationEventSuccessfully();
+
+        consumer.commitSync(mockTopicPartitionOffset());
+
+        assertEquals(1, MockConsumerInterceptor.ON_COMMIT_COUNT.get());
+    }
+
+    @Test
+    public void testInterceptorCommitAsync() {
+        Properties props = requiredConsumerPropertiesAndGroupId("test-id");
+        props.setProperty(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, MockConsumerInterceptor.class.getName());
+        props.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+        consumer = newConsumer(props);
+        assertEquals(1, MockConsumerInterceptor.INIT_COUNT.get());
+
+        completeCommitApplicationEventSuccessfully();
+        consumer.commitAsync(mockTopicPartitionOffset(), new MockCommitCallback());
+        assertEquals(0, MockConsumerInterceptor.ON_COMMIT_COUNT.get());
+
+        forceCommitCallbackInvocation();
+        assertEquals(1, MockConsumerInterceptor.ON_COMMIT_COUNT.get());
+    }
+
+    @Test
+    public void testNoInterceptorCommitAsyncFailed() {
```

Comment:
Is there a reason why we do not have the corresponding test for `commitSync()`?

Reply:
I added one.